src/main/java/com/pi4j/driver/sensor/bmx280/Bmx280Driver.java

```diff
@@ -0,0 +1,420 @@
+            var2 = var2 + var1 * digP5 * 2.0;
+            var2 = (var2 / 4.0) + (digP4 * 65536.0);
+            var1 = (digP3 * var1 * var1 / 524288.0 + digP2 * var1) / 524288.0;
+            var1 = (1.0 + var1 / 32768.0) * digP1;
+            if (var1 == 0.0) {
+                pressure = 0;   // // avoid exception caused by division by zero
+            } else {
+                pressure = 1048576.0 - adcP;
+                pressure = (pressure - (var2 / 4096.0)) * 6250.0 / var1;
+                var1 = digP9 * pressure * pressure / 2147483648.0;
+                var2 = pressure * digP8 / 32768.0;
+                pressure = pressure + (var1 + var2 + digP7) / 16.0;
+           }
+        }
+        float adcH = Float.NaN;
+        double humidity = Double.NaN;
+        if (sensorType == SensorType.BME280 && humidityMode != SensorMode.DISABLED) {
+            // Humidity
+
+            adcH = ((ioBuf[6] & 0xFF) << 8) | (ioBuf[7] & 0xFF);
+
+            double varH = tFine - 76800.0;
+            varH = (adcH - (digH4 * 64.0 + digH5 / 16384.0 *
+                    varH)) * (digH2 / 65536.0 * (1.0 + digH6 /
+                    67108864.0 * varH *
+                    (1.0 + digH3 / 67108864.0 * varH)));
+            varH = varH * (1.0 - digH1 * varH / 524288.0);
+
+            if (varH > 100.0) {
+                varH = 100.0;
+            } else if (varH < 0.0) {
+                varH = 0.0;
+            }
+            humidity = varH;
+        }
+
+        if (measurementMode == MeasurementMode.FORCED) {
+            measurementMode = MeasurementMode.SLEEPING;
+        }
+
+        return new Measurement(
+                adcT / 1024.0f, adcP / 1024.0f, adcH / 1024.0f,
+                (float) temperature, (float) pressure, (float) humidity);
+    }
+
+    /**
+     * Write the reset command to the BMP280.
+     */
+    public void reset() {
+        materializeSleep(false);
+        registerAccess.writeRegister(Bmp280Constants.RESET, Bmp280Constants.RESET_CMD);
+        busyUntil = System.currentTimeMillis() + 100;
+    }
+
+    public SensorType getSensorType() {
+        return sensorType;
+    }
+
+    // Internal methods
+
+    private void materializeSleep(boolean forMeasurement) {
+        long timeOut = forMeasurement ? Math.max(busyUntil, measurementAvailableAt) : busyUntil;
+        while (true) {
+            long now = System.currentTimeMillis();
+            if (now >= timeOut) {
+                return;
+            }
+            try {
+                Thread.sleep(timeOut - now);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    private int readRegisterS16(int register) {
+        registerAccess.readRegister(register, ioBuf, 2);
+        return (ioBuf[0] & 0xFF) | (ioBuf[1] << 8);
+    }
+
+    private int readRegisterU16(int register) {
+        return readRegisterS16(register) & 0xFFFF;
+    }
+
+    // Nested types
```

Comment:
Personal opinion: I prefer not to have nested types, but e.g. a package with all the enums as separate classes.

Reply:
PS to keep this going perhaps let's have a preference poll for the various options?